File: VB/DomainComponents/Interfaces.vb
Imports DevExpress.ExpressApp.DC

Namespace DomainComponents
    <DomainComponent> _
    Public Interface IPerson
        Property Name() As String
    End Interface
    <DomainComponent> _
    Public Interface IProduct
        Property Price() As Decimal
        Property Name() As String
    End Interface
    <DomainComponent> _
    Public Interface ISale
        Property Product() As IProduct
        Property Person() As IPerson
        Property Count() As Decimal
    End Interface
End Namespace

File: CS/StandaloneDC/DomainComponents/Interfaces.cs
using System;
using DevExpress.ExpressApp.DC;

namespace DomainComponents {
    [DomainComponent]
    public interface IPerson {
        String Name { get; set; }        
    }
    [DomainComponent]
    public interface IProduct {
        Decimal Price { get; set; }
        String Name { get; set; }        
    }
    [DomainComponent]
    public interface ISale {
        IProduct Product { get; set; }
        IPerson Person { get; set; }
        Decimal Count { get; set; }
    }
}

 

